csrc/tensor_view.cpp

```diff
@@ -1168,15 +1168,26 @@ TensorView* TensorView::cacheFork() {
 TensorView* TensorView::cacheAfter(
     LoadStoreOpType op_type,
     CacheOp cache_op,
-    bool propagate_allocation_domain) {
+    bool propagate_allocation_domain,
+    const std::unordered_set<Expr*>& cached_uses) {
   NVF_ERROR(
       !container()->isA<kir::Kernel>(),
       "Function invalid for kernel container.");
   FusionGuard fg(fusion());
 
+  auto target_uses = fusion()->unordered_uses(this);
```

Comment:
It's used for mostly checks, except
```
  for (auto expr : target_uses) {
    ir_utils::replaceValInExprInputs(expr, this, consumer);
  }
```

I can switch to `uses()` and filter it by uniqueness.

Reply:
Yes, I'm concerned about that part since `replaceValInExprInputs` would create new IR nodes, and while the final kernel would do the same computations, the generated kernel would have non-deterministic variable names.